crates/ruff_linter/resources/test/fixtures/airflow/AIR302_context.py

```diff
@@ -0,0 +1,19 @@
+from airflow.decorators import task
+
+@task
+def print_config(**context):
+    # This should not throw an error as logical_date is part of airflow context.
+    logical_date = context["logical_date"]
```

Comment:
@sunank200 and I discussed this earlier. What we're trying to check is whether there's a variable named as `context` in a function (most commonly seen in taskflow and python operator) and whether it's can be accessed like a dict with the keys we want to check. I think it's unlikely users are using something like this out of the airflow context. But would like to know whether there's any concern

@MichaReiser @uranusjr

Reply:
I have added logic for other ways to access context value as well. It is part of tests.

Reply:
It’s probably better to detect

1. Arguments of a function decorated with `@task` (either `**` or simple named arguments). (As a follow-up, any functions called by such a function)
2. The `execute` function of a BaseOperator subclass (As a follow-up, any functions called by `execute`)
3. The dict returned by `get_current_context`.

This should be better than detecting with variable name.

Reply:
What about `python_callable`?

Reply:
I don’t think `python_callable` takes the context though? It only accepts things you provide in `self.op_args` and `self.op_kwargs`.

Reply:
I agree that it'll be useful to guard this check by first verifying that the parameter is coming from a function which is decorated with a `@task`.

I think this can be done as a pre-check for context variables by using the `checker.semantic().current_statements()` method to traverse up the AST to find the function definition node and checking whether the function has a `@task` decorator that originates from the `airflow` module.

https://github.com/astral-sh/ruff/blob/9fd4eb8c4cd3ffe4de0540f368a8350a0f920e07/crates/ruff_python_semantic/src/model.rs#L1232-L1239

Reply:
> I don’t think `python_callable` takes the context though? It only accepts things you provide in `self.op_args` and `self.op_kwargs`.

I though we can still get it in the python_callable? https://airflow.apache.org/docs/apache-airflow/stable/howto/operator/python.html#pythonoperator